cmake/scripts/darwin_embedded/Macros.cmake

```diff
@@ -2,7 +2,9 @@ function(core_link_library lib wraplib)
   if(CMAKE_GENERATOR MATCHES "Unix Makefiles" OR CMAKE_GENERATOR STREQUAL Ninja)
     set(wrapper_obj cores/dll-loader/exports/CMakeFiles/wrapper.dir/wrapper.c.o)
   elseif(CMAKE_GENERATOR MATCHES "Xcode")
-    set(wrapper_obj cores/dll-loader/exports/kodi.build/$(CONFIGURATION)$(EFFECTIVE_PLATFORM_NAME)/wrapper.build/Objects-$(CURRENT_VARIANT)/$(CURRENT_ARCH)/wrapper.o)
+    # CURRENT_VARIANT is an Xcode env var
+    # CPU is a project cmake var
+    set(wrapper_obj cores/dll-loader/exports/kodi.build/$(CONFIGURATION)$(EFFECTIVE_PLATFORM_NAME)/wrapper.build/Objects-$CURRENT_VARIANT/${CPU}/wrapper.o)
```

Comment:
why removing parentheses in `CURRENT_VARIANT`? (also in osx)

Reply:
yeah, so thats the quoting issue i allude to. for an xcode buildphase script, the xcode env vars are accessible via $VARIABLE_NAME. however, configuration and effective_platform_name are actually unique in that cmake explicitly check for $(CONFIGURATION) and strip and pass it as $CONFIGURATION into the generated script. No other env var is handled this way for whatever reason. Cmake does some weird escaping at times (cmake issue for reference/explanation https://gitlab.kitware.com/cmake/cmake/-/issues/18062)

For some reason the new xcode build system fails (where the old one doesnt care). 

Also the change to CPU is because CURRENT_ARCH is no longer available. ARCHS can be a string of multiple (eg x86_64:arm64), and the wrap_libdvdnav target doesnt have an arch so we just have to rely on our cmake var for that.

Reply:
that's quite some black magic involved... Thank you for the detailed explanation!